sdk/appconfiguration/app-configuration/test/internal/http.spec.ts

```diff
@@ -138,160 +139,162 @@ describe("http request related tests", function () {
   // they send or reply back with).
-  describe("request/reply tests for sync token headers", () => {
-    let client: AppConfigurationClient;
-    let syncTokens: SyncTokens;
-    let scope: nock.Scope;
-
-    beforeEach(function (this: Context) {
-      if (nock == null || nock.recorder == null) {
-        this.skip();
-        return;
-      }
-
-      syncTokens = new SyncTokens();
-
-      client =
-        createAppConfigurationClientForTests<InternalAppConfigurationClientOptions>({
-          syncTokens: syncTokens,
-        }) || this.skip();
-
-      nock.recorder.clear();
-      nock.restore();
-      nock.cleanAll();
-      if (!nock.isActive()) {
-        nock.activate();
-      }
-      scope = nock(/.*/);
-    });
-
-    afterEach(function (this: Context) {
-      if (nock == null || nock.recorder == null) {
-        return;
-      }
-
-      if (!this.currentTest?.isPending()) {
-        assert.ok(scope.isDone());
-      }
-      nock.recorder.clear();
-      nock.restore();
-      nock.cleanAll();
-    });
-
-    it("policy is setup properly to send sync tokens", async function () {
-      syncTokens.addSyncTokenFromHeaderValue(`hello=world;sn=1`);
-
-      const policyScope = nock(/.*/, {
-        reqheaders: {
-          "sync-token": "hello=world",
-        },
-      })
-        .get(/.*/)
-        .reply(418);
-
-      await assertThrowsRestError(
-        async () =>
-          client.getConfigurationSetting({
-            key: "doesntmatter",
-          }),
-        418
-      );
-
-      assert.ok(policyScope.isDone());
-    });
-
-    it("addConfigurationSetting", async () => {
-      scope.put(/.*/).reply(200, "", { "sync-token": "addConfigurationSetting=value;sn=1" });
-
-      await client.addConfigurationSetting({
-        key: "doesntmatter",
-      });
-
-      assert.equal(syncTokens.getSyncTokenHeaderValue(), "addConfigurationSetting=value");
-    });
-
-    it("getConfigurationSetting", async () => {
-      scope.get(/.*/).reply(200, "", { "sync-token": "getConfigurationSetting=value;sn=1" });
-
-      await client.getConfigurationSetting({
-        key: "doesntmatter",
-      });
-
-      assert.equal(syncTokens.getSyncTokenHeaderValue(), "getConfigurationSetting=value");
-    });
-
-    it("setConfigurationSetting", async () => {
-      scope.put(/.*/).reply(200, "", { "sync-token": "setConfigurationSetting=value;sn=1" });
+  // describe("request/reply tests for sync token headers", () => {
+  //   let client: AppConfigurationClient;
+  //   let syncTokens: SyncTokens;
+  //   let scope: nock.Scope;
+
+  //   beforeEach(function(this: Context) {
+  //     if (nock == null || nock.recorder == null) {
+  //       this.skip();
+  //       return;
+  //     }
+
+  //     syncTokens = new SyncTokens();
+
+  //     client =
+  //       createAppConfigurationClientForTests<InternalAppConfigurationClientOptions>({
+  //         syncTokens: syncTokens
+  //       }) || this.skip();
+
+  //     nock.recorder.clear();
```

Comment:
Could you elaborate on why these tests interface with `nock` directly and whether/what the alternative is for this use case with the new recorder?

Reply:
Richard added those tests, I need to review them again to see if I should keep it or remove.